### PR TITLE
Admin view

### PR DESCRIPTION
**Pull request by HedicGuibert**
This PR fixes [#85](https://github.com/jolicode/starfleet/issues/85#issue-750847364) .

It adds createdAt and speakers columns to the admin talk view. It also adds a talks column to the user list view, but this may be unnecessary.

Talks list view :
Talks

Users list view :
EDIT :  removed

Users

### CODE REVIEW

**Comment by lyrixx** [comment]
Please, add some screenshot. Thanks

**Comment by HedicGuibert** [comment]
Sure, sorry for this omission.
I updated my innitial comment.

**Comment by welcoMattic** [comment]
I'm afraid that the talks column on the user list view is not necessary. In fact, the talk list of an user will increase over time, so the content of this column to, it will quickly become unreadable.

File: src/DataFixtures/TalkFixtures.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+use App\Entity\Talk;
+use App\Entity\User;
+use Doctrine\Bundle\FixturesBundle\Fixture;
+use Doctrine\Persistence\ObjectManager;
+
+class TalkFixtures extends Fixture
+{
+    public function load(ObjectManager $manager)
+    {
+
```
Comment (lyrixx) [inline src/DataFixtures/TalkFixtures.php:15]: useless space

**Comment by HedicGuibert** [comment]
Ok, this makes sense, I'm removing it then

File: src/Entity/Talk.php [hunk 2]
```diff
@@ -46,9 +46,15 @@ class Talk
      */
     private $submits;
 
+    /**
+     * @ORM\OneToMany(targetEntity=User::class, mappedBy="talks")
+     */
+    private $authors;
```
Comment (lyrixx) [inline src/Entity/Talk.php:52]: Why did you add this new property? more over this seems wrong (never updated, never used)

**Review by lyrixx — CHANGES_REQUESTED** [review]

File: src/Entity/Talk.php (hunk 2 above)
Comment (welcoMattic) [inline src/Entity/Talk.php:52]: Hmm, I think there is a design issue here. Actually the data schema is:
User 1->n Submit n->1 Talk
Here, you are adding a direct relation between User and Talk, but this relation exists, indirectly through Submit. 
So, you might be able to retrieve all talks of a user through there submits, with something like:

```
    public function getAuthors(): string
    {
        return array_reduce($this->getSubmits(), function ($r, Submit $submit) {
            foreach ($submit->getUsers() as $user) {
                $r = '' === $r ? $user->getName() : $r . ', ' . $user->getName();
            }

            return $r;
        }, '');
    }
```

(I did not test this code)

Comment (HedicGuibert) [reply src/Entity/Talk.php:52]: Isn't a relation needed between Talk and User to allow the speakers to be displayed on the talks list view ?

Comment (HedicGuibert) [reply src/Entity/Talk.php:52]: Oh sure, it looks a lot better indeed and adresses what is said above.
Thanks !

Comment (lyrixx) [reply src/Entity/Talk.php:52]: And BTW, the code here is ugly. Use a template in easy admin for that

File: templates/easy_admin/Talk/authors.html.twig [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{% for submit in value %}
+    {% for user in submit.users %}
```
Comment (lyrixx) [inline templates/easy_admin/Talk/authors.html.twig:2]: I would use `<ul><li>` here ?

File: config/packages/easy_admin/talk.yaml [hunk 4]
```diff
@@ -7,6 +7,8 @@ easy_admin:
                 fields:
                     - { property: 'title', type: 'text' }
                     - { property: 'submits' }
+                    - { property: 'createdAt', type: 'date' }
+                    - { property: 'authors', type: 'string', template: 'easy_admin/Talk/authors.html.twig' }
```
Comment (lyrixx) [inline config/packages/easy_admin/talk.yaml:11]: `Talk` -> `talk`

File: src/Entity/Talk.php [hunk 5]
```diff
@@ -101,6 +101,11 @@ public function removeSubmit(Submit $submit): self
         return $this;
     }
 
+    public function getAuthors()
+    {
+        return $this->getSubmits();
```
Comment (lyrixx) [inline src/Entity/Talk.php:106]: The wording seems odd Author VS Submits, and anyway, I don't think this method is useful anyway :)

**Comment by lyrixx** [comment]
Could you also sqash your commits, because it does not say what they do

File: config/packages/easy_admin/talk.yaml [hunk 6]
```diff
@@ -7,6 +7,8 @@ easy_admin:
                 fields:
                     - { property: 'title', type: 'text' }
                     - { property: 'submits' }
+                    - { property: 'createdAt', type: 'date' }
+                    - { property: 'authors', type: 'text', template: 'easy_admin/talk/authors.html.twig' }
```
Comment (lyrixx) [inline config/packages/easy_admin/talk.yaml:11]: The type here is useless since you are using a template

File: templates/easy_admin/talk/authors.html.twig [hunk 7]
```diff
@@ -0,0 +1,7 @@
+{% for submit in item.submits %}
+    <ul style="margin-bottom:0">
```
Comment (lyrixx) [inline templates/easy_admin/talk/authors.html.twig:2]: I would put the `ul` before the first for, or you'll get many ul in this table cell.
And IMHO, we will have lot's of duplicate content here.
If a user submit 5 talks. He'll appear here 5 times.
You should unique the list

Comment (welcoMattic) [reply templates/easy_admin/talk/authors.html.twig:2]: Even if most of time there will be 1 speaker, if a talk is submitted 2 times with 2 different speakers couple (let say Alice and Bob, and Alice and Jack), we will see "Alice, Bob, Jack", which is not very relevant, don't you think? 
May be unique the list, but "by submit"
1st submit: Alice
2nd submit: Alice
3rd submit: Alice and Bob
4th submit: Alice and Jack
5th submit: Alice and Bob
will result to:
Alice, Alice + Bob, Alice + Jack

WDYT?

**Comment by HedicGuibert** [comment]
I updated the view accordingly to what you said, and I added some fixtures.
Here is what it looks like now :
NewTalksView

**Review by lyrixx — APPROVED** [review]
I approve, even if you did not uniq the list. That's an edge and it seems you did not manage to succeed. Not a big deal.

**Comment by HedicGuibert** [comment]
I did it locally but I felt like it wasn't useful, seems I was wrong ? I can still add it
